Dockerfile

```diff
@@ -0,0 +1,25 @@
+FROM golang:1.16 AS mod_cacher
+
+WORKDIR /app
+
+COPY go.mod go.sum /app/
+RUN go mod download
+
+FROM golang:1.16 AS builder
```

Comment:
@miraclesu may i know the reason you would like to use separated images for go mod, executable builder and runner?

Reply:
Generally, the changes of the mod file are relatively few compared to the code.
So, we can cache the mod packages instead of re-downloading the mod every time `docker run` after changes the code.

Reply:
Understood. Can try in this way first.